subsys/bluetooth/host/conn.c

```diff
@@ -2614,16 +2614,175 @@ static int bt_conn_get_tx_power_level(struct bt_conn *conn, uint8_t type,
 	return 0;
 }
 
+#if defined(CONFIG_BT_TRANSMIT_POWER_CONTROL)
+void notify_tx_power_report(struct bt_conn *conn,
+			    struct bt_conn_le_tx_power_report report)
+{
+	for (struct bt_conn_cb *cb = callback_list; cb; cb = cb->_next) {
+		if (cb->tx_power_report) {
+			cb->tx_power_report(conn, &report);
+		}
+	}
+
+	STRUCT_SECTION_FOREACH(bt_conn_cb, cb)
+	{
+		if (cb->tx_power_report) {
+			cb->tx_power_report(conn, &report);
+		}
+	}
+
+	bt_conn_unref(conn);
```

Comment:
This is incorrect. As a rule of thumb, functions should not move references.

Eg: The reference in `conn` is borrowed here from `conn` in `bt_hci_le_transmit_power_report`. An unref here is a double free.

```suggestion
```

Reply:
Thanks! :) Will change.